src/Trace/Trace.php

```diff
@@ -0,0 +1,158 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Trace;
+
+/**
+ * This plain PHP class represents a Trace resource. Traces belong to a
+ * project and have many TraceSpans. See:
+ * https://cloud.google.com/trace/docs/reference/v1/rest/v1/projects.traces#resource-trace
+ *
+ *
+ * Example:
+ * ```
+ * use Google\Cloud\ServiceBuilder;
+ *
+ * $cloud = new ServiceBuilder();
+ * $traceClient = $cloud->trace();
+ *
+ * $trace = $traceClient->trace();
+ * ```
+ */
+class Trace
+{
+    /**
+     * The id of the project this trace belongs to.
+     * @var string
+     */
+    private $projectId;
+
+    /**
+     * The trace id for this trace. 128-bit numeric formatted as a 32-byte hex string
+     * @var string
+     */
+    private $traceId;
+
+    /**
+     * List of TraceSpans to report
+     * @var TraceSpan[]
+     */
+    private $spans = [];
+
+    /**
+     * Instantiate a new Trace instance.
+     *
+     * @param array $options {
+     *      Configuration options.
+     *
+     *      @type string $projectId The id of the project this trace belongs to.
+     *      @type string $traceId [optional] The trace id for this trace. 128-bit numeric
+     *            formatted as a 32-byte hex string. If not provided, one will be generated
+     *            automatically for you.
+     *      @type array $spans [optional] List of span data to load.
+     * }
+     */
+    public function __construct(array $options)
+    {
+        $this->projectId = $options['projectId'];
+
+        if (array_key_exists('traceId', $options)) {
+            $this->traceId = $options['traceId'];
+        }
+        $this->traceId = $this->traceId ?: $this->generateTraceId();
+
+        if (array_key_exists('spans', $options)) {
+            foreach ($options['spans'] as $span) {
+                array_push($this->spans, new TraceSpan($span));
+            }
+        }
+    }
+
+    /**
+     * Retrieves the trace's id.
+     *
+     * @return string
+     */
+    public function getTraceId()
+    {
+        return $this->traceId;
+    }
+
+    /**
+     * Returns a serializable array representing this trace.
+     *
+     * @return array
+     */
+    public function info()
+    {
+        return [
+            'projectId' => $this->projectId,
+            'traceId' => $this->traceId,
+            'spans' => array_map(function ($span) {
+                return $span->info();
+            }, $this->spans)
+        ];
+    }
+
+    /**
+     * Retrieves the spans for this trace.
+     *
+     * @return array
+     */
+    public function getSpans()
+    {
+        return $this->spans;
+    }
+
+    /**
+     * Set the spans for this trace.
+     *
+     * @param TraceSpan[] $spans
+     */
+    public function setSpans(array $spans)
```

Comment:
You can use `Google\Cloud\ValidateTrait` to ensure that `$spans` contains only instances of `TraceSpan`.

```php
use ValidateTrait;

$this->validateBatch($spans, TraceSpan::class);
```